libkbfs/block_types.go

```diff
@@ -132,35 +129,34 @@ func (db *DirBlock) NewEmpty() Block {
 }
 
 // Set implements the Block interface for DirBlock
-func (db *DirBlock) Set(other Block, codec kbfscodec.Codec) {
+func (db *DirBlock) Set(other Block) {
 	otherDb := other.(*DirBlock)
-	dbCopy, err := otherDb.DeepCopy(codec)
-	if err != nil {
-		panic("Unable to DirBlock.Set")
-	}
-	db.IsInd = dbCopy.IsInd
-	db.UnknownFieldSetHandler = dbCopy.UnknownFieldSetHandler
+	dbCopy := otherDb.DeepCopy()
 	db.Children = dbCopy.Children
 	db.IPtrs = dbCopy.IPtrs
-	db.cacheMtx.Lock()
-	defer db.cacheMtx.Unlock()
-	db.cachedEncodedSize = dbCopy.cachedEncodedSize
+	db.CommonBlock.IsInd = dbCopy.IsInd
```

Comment:
Does `db.CommonBlock.Set(dbCopy.CommonBlock)` not work?  (Or maybe `db.ToCommonBlock().Set(dbCopy.CommonBlock)`.)

Even if not, you wouldn't need the `.CommonBlock` part of these three lines, right?